Nonprofit-Exchange-Hub/web-app
# #209 [WIP] user images step 2

# Important
 **blocked by: https://github.com/Nonprofit-Exchange-Hub/web-app/pull/264**
## Dev Summary
# Not ready for review. This work is blocked by 
### Changes
- [x] Refactored `AuthModule` and `UsersModule` into a single account manager module since these play a similar role and are usually imported together to get any auth functionality, it make sense to keep them together. It makes testing easier: https://blog.logrocket.com/avoid-circular-dependencies-nestjs/
- [x] Added a top level `FileService` that implements Azure blob storage for persisting files. The caveat with this is that there is still work to be done to replace all files, since files are saved with an extension, there could be x number of extension possible files to check.
- [x] Added an avatar component to the UI that renders the file if the user has a profile image
- [x] Added profile image to the global `User` type in react  

### not implemented
implementation for local dev like saving to local dev environment, maybe later


## Test Plan

repro steps:

1. Start the app
2. using an http client like insomnia, log into the route: `http://localhost:3001/api/auth/login`, when successful this will save the cookie
3. Make a post request to `http://localhost:3001/api/auth/profile/{userid}` with multipart form body:
  - include a form property `profile_image_url` and use insomnia or postman to include an image file in the form request (smaller than 1MB)

Before testing, contact me to get the Azure api key
Expect to see the image as an avatar for the user when you log in as the same user in react. The file will be saved a public Azure blob (inspect the image url)



## Resources

- [stackoverflow is cool](stackoverflow.com)
